src/js/utils/template-processor.js

```diff
@@ -23,33 +23,47 @@ const operators = {
   '??': (value, defaults) => value ?? defaults,
 };
 
-class PartialDirective {
+class PartialTemplate {
   constructor(template) {
     this.template = template;
   }
 }
 
+const Directives = {
+  partial: (expression, part, state) => {
+    state[expression] = new PartialTemplate(part.template);
+  },
+  if: (expression, part, state) => {
+    if (evaluateCondition(part.expression, state)) {
+      part.replace(new TemplateInstance(part.template, state, processor));
+    } else {
+      part.replace('');
+    }
+  },
+};
+
+const DirectiveNames = Object.keys(Directives);
+
+export function transformDirectiveAliases(template) {
```

Comment:
suggestion(non-blocking): I think `transform` functions tend to indicate (still potentially mutative) new objects (e.g. https://lodash.com/docs/4.17.15#transform or https://www.educative.io/answers/what-is-the-transform-function-in-cpp). Consider calling this `updateDirectiveAliases` or `modifyDirectAliases` or something like that? (I dunno, names are hard)